src/components/Dropdown/DropdownBasic.tsx

```diff
@@ -52,7 +53,49 @@ export const DropdownBasic = (props: IDropdownBasicProps): ReactElement => {
     setVisible(false);
   };
 
-  const containerClasses = clsx('z-50 bg-white', classes.container, {
+  const open = () => {
+    visible || setVisible(true);
```

Comment:
I think you want to do an `&&` here, since the `||` is going to only ever run `setVisible()` when `visible === false`

We can change this code, as below:

```js
// helpers for setting visibility
const close = () => visible && setVisible(false);
const open = () => !visible && setVisible(true);
```

Reply:
Technically that's true, on second look at this! :laughing:  
I would say, looking at this again we should scrap doing this type of short-circuiting since React will optimize a state update anyway.
We can just do `open = () => setVisible(true);` and if its already `true` react won't re-render.

Reply:
Updated